crates/telio-traversal/src/last_handshake_time_provider.rs

```diff
@@ -29,13 +29,8 @@ pub struct WireGuardLastHandshakeTimeProvider {
 #[async_trait]
 impl LastHandshakeTimeProvider for WireGuardLastHandshakeTimeProvider {
     async fn last_handshake_time(&self, public_key: &PublicKey) -> Result<Option<Duration>, Error> {
```

Comment:
That is true, somehow I just kept whatever it was before, but your point is very logical.

Let me update that.

Reply:
Ok, so there is a bit of reason for having `Result<Option<Duration>>` as `Err(..)` in this case indicates a failure to retrieve the time, but on the other hand, it is possible, that retrieval is succesfull, but the peers have not simply communicated yet. In that case it would be nicer to have `Ok(None)` instead of `Err(...)` as it is not an error condition. 

But I will change slightly how error mapping is done to make it more concise... And actually allow for None value.

Reply:
:heavy_check_mark: